client/extensions/woocommerce/components/address-view/index.js

```diff
@@ -1,51 +1,187 @@
 /**
  * External dependencies
  */
+import classNames from 'classnames';
+import debugFactory from 'debug';
+import { localize } from 'i18n-calypso';
 import React, { Component, PropTypes } from 'react';
-import i18n from 'i18n-calypso';
+
+const debug = debugFactory( 'calypso:allendav' );
 
 /**
  * Internal dependencies
  */
-class AddressView extends Component {
+import Countries from 'woocommerce/lib/countries';
+import FormFieldSet from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormSelect from 'components/forms/form-select';
+import FormTextInput from 'components/forms/form-text-input';
 
+class AddressView extends Component {
 	static propTypes = {
 		address: PropTypes.shape( {
 			name: PropTypes.string.isRequired,
 			street: PropTypes.string.isRequired,
+			street2: PropTypes.string,
 			city: PropTypes.string.isRequired,
+			state: PropTypes.string,
 			country: PropTypes.string.isRequired,
+			postcode: PropTypes.string,
 		} ),
+		allowedCountries: PropTypes.arrayOf( PropTypes.string ),
+		isEditable: PropTypes.bool,
+		nameLabel: PropTypes.string,
+		onChange: PropTypes.func,
+		onFocus: PropTypes.func,
 	};
 
+	static defaultProps = {
+		address: {
+			name: '',
+			street: '',
+			street2: '',
+			city: '',
+			state: '',
+			country: 'US',
+			postcode: '',
+		},
+		allowedCountries: [
+			'US', 'CA'
+		],
+		isEditable: false,
+		nameLabel: '',
+	}
+
 	constructor( props ) {
 		super( props );
 
-		this.edit = this.edit.bind( this );
+		this.state = {
+			address: props.address,
+		};
 	}
 
-	edit() {
-		//TODO: Add edit functionality
-		return false;
+	getStateFieldLabel = () => {
+		const { country, translate } = this.props;
+
+		if ( 'CA' === country ) {
+			return translate( 'Province' );
```

Comment:
Hi! I've found a possible matching string that has already been translated **58** times:
`translate( 'State/Province' )` *ES Score: 11*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).